corehq/apps/hqcase/tests/test_case_list_api.py

```diff
@@ -0,0 +1,138 @@
+                ('chaney', "Tom Chaney", {"alias": "The Coward"}, bad_id),
+                ('ned', "Ned Pepper", {"alias": "Lucky Ned"}, bad_id),
+        ]:
+            case_blocks.append(CaseBlock(
+                case_id=str(uuid.uuid4()),
+                case_type='person',
+                case_name=name,
+                external_id=external_id,
+                owner_id='person_owner',
+                date_opened=date_opened,
+                create=True,
+                update=properties,
+                index={'parent': IndexAttrs('team', team_id, 'child')}
+            ))
+            date_opened += datetime.timedelta(days=1)
+
+        case_blocks[-1].close = True  # close Ned Pepper
+
+        _, cases = submit_case_blocks([cb.as_text() for cb in case_blocks], domain=cls.domain)
+
+        # preserve ordering so inserted_at date lines up right in ES
+        order = {cb.external_id: index for index, cb in enumerate(case_blocks)}
+        return sorted(cases, key=lambda case: order[case.external_id])
+
+    @classmethod
+    def tearDownClass(cls):
+        FormProcessorTestUtils.delete_all_cases()
+        ensure_index_deleted(CASE_SEARCH_INDEX_INFO.index)
+        super().tearDownClass()
+
+
+@generate_cases([
+    ("", ['good_guys', 'bad_guys', 'mattie', 'rooster', 'laboeuf', 'chaney', 'ned']),
+    ("limit=2", ['good_guys', 'bad_guys']),
+    ("external_id=mattie", ['mattie']),
+    ("external_id=the-man-with-no-name", []),
+    ("case_type=team", ["good_guys", "bad_guys"]),
+    ("owner_id=person_owner", ['mattie', 'rooster', 'laboeuf', 'chaney', 'ned']),
+    ("case_name=Mattie Ross", ["mattie"]),
+    ("case_name=Mattie+Ross", ["mattie"]),
+    ("case_name=Mattie", []),
+    ("closed=true", ["ned"]),
+    ("date_opened.lt=1878-02-19", ["mattie", "rooster"]),
+    ("date_opened.lte=1878-02-19", ["mattie", "rooster", "laboeuf"]),
+    ("date_opened.lte=1878-02-19T00:00:00", ["mattie", "rooster"]),
+    ("date_opened.gt=1878-02-18&date_opened.lt=1878-02-20", ["laboeuf"]),
+    ("date_opened.gt=1878-02-19T11:00:00&date_opened.lt=1878-02-19T13:00:00", ["laboeuf"]),
+    ("date_opened.lt=1878-02-18&date_opened.gt=1878-02-19", []),
+    ("property.alias=Rooster", ["rooster"]),
+    ("property.alias=rooster", []),
+    ('property.foo {"test": "json"}=bar', []),  # This is escaped as expected
+    ('property.foo={"test": "json"}', []),  # This is escaped as expected
+    ("case_type=person&property.alias=", ["mattie", "laboeuf"]),
+], TestCaseListAPI)
+def test_case_list_queries(self, querystring, expected):
+    params = QueryDict(querystring).dict()
+    actual = [c['external_id'] for c in get_list(self.domain, params)]
+    # order matters, so this doesn't use assertItemsEqual
+    self.assertEqual(actual, expected)
+
+
+@generate_cases([
+    ("limit=nolimitz", "'nolimitz' is not a valid value for 'limit'"),
+    ("limit=10000", "You cannot request more than 5000 cases per request."),
```

Comment:
should you use the `MAX_PAGE_SIZE` const here? Otherwise this check will fail if you ever change that.